packages/core/src/http-client/http-client-types.ts

```diff
@@ -24,11 +43,15 @@ export enum HttpMethod {
  */
 export interface HttpRequestConfig {
   url?: string;
-  method: HttpMethod;
-  data?: MapType<any>;
+  method: Method;
```

Comment:
Adding on to what Frank said, isn't this a breaking change? Can you provide some background on why you changed this?

Reply:
See above. It is not breaking because you can still use the old enum. I don't know why decided to use that enum in the first place, but it does not make any sense to me, when there is a more straight forward option.

Reply:
I think there was a reason for that, I will get back to you.

Reply:
Okay I think I found out again:

If you use the API like this, everything works fine:

```ts
executeHttpRequest(destination, {
  method: 'get',
  // more stuff
});
```

If you use it like this, however, tsc will complain:

```ts
const config = {
  method: 'get',
  // more stuff
};

executeHttpRequest(destination, config);
```

tsc will complain that `Type 'string' is not assignable to type 'Method'.ts(2345)`. I think that is why we decided to use an enum. To be fair, we should have documented that somewhere, but apparently missed that.

Reply:
The question is, which is worse?

Reply:
I quickly tried this in [a playground](https://www.typescriptlang.org/play/#code/C4TwDgpgBAshwAsD2ATKBeKByA5vLUAPtgOICiAKgcVmEgM7DXYAKA8gMpUDcAULwEsAdsAgAnAGYBDAMbQAShACOUAN68oUALbxkKAFyxdqPgF9+EgK5CZwAUiFQIADwgzLogBRjlhxUoBKNX5zXgB6MKgZKXpoAEZDS1ioJAAjACs3YCgUAR9bABsQXhc3DwhPVW1jA2w8JihTAL5wyOjkgCZDYQE7KQKBAC9oNMzbKFSICSQfXhkHRigfJTiMNWrEVENcfEa+UvcvZbjm-giomOgAZkMpFDRQSCgpISEkYCk7BzmF7OWuqD+NZVHSbWo7BqmfauQ4Vf6nVoXZIAFlu9FiYmyMSivx+QkWyyuwI2em29QI2Pm+OyUJKMPK3mUV2aQA) and to my understanding this is due to the fact, that the "content" of a (const) object is not necessarily const. There are multiple ways (in addition to the one you mentioned) to solve this, one of them being that you can assert it to be const.

Reply:
Interesting, I didn't event know that `as const` is a thing! At the end of the day, it's the same as `config: HttpRequestConfig` or `as Method` in green, isn't it? At least `as const` and `as Method` aren't things I'd do if I didn't have to, I'd only do it so that tsc stops complaining. Adding explicit type annotations is a matter of taste, I guess, but it's nothing I personally tend to do.
I extended your playground a little and it seems like you're correct in that it's not a breaking change. As long as the all members of the enum are also part of the union type, existing code shouldn't break. 
That makes the whole thing a subjective decision. Personally I still prefer the enum, since I favor the consistency over the convenience here. If you disagree, should we just let Frank decide? Then it's 2 v 1 in either direction.

Reply:
I guess so. I still prefer the union type and I would add, that in my opinion this is not only more convenient but also more consistent, because we claim that you can pass a valid axios config to our generic http-client, but in fact we do not support all configs. Most examples you find for axios use lowercase, but we only allow uppercase. I find that confusing.

Reply:
Sry, unfortunate phrasing: With consistent I mean "works both as reference and as object literal". Our claim that a valid Axios config can be passed is actually incorrect, but apparently no one noticed so far 😄 @FrankEssenberger your input would be appreciated here

Reply:
First a remark to your playground finding you could also help the poor compiler by adding the type to the config so he/she/it knows it is not a plain object: 

```
const config:HttpRequestConfig = {method:'get'}
executeHttpRequest({url:'foo.bar'},config)
```

The option using the `as Method` I would totally avoid because it destroys any safety. Then you can pass any string to the executor.

 In general I have the following feeling:
- Personally I am more on the enum bus because I prefer to have no strings at all in the code. But this is just me and a bit the Java background.
- In the TS context I would say that string union types are a slim alternative providing the same degree of safety but are also more in line with the natural key/value of JSON and in the TS/JS world you have strings all over the place so it seems fine.

#Conclusion
I would also go for string union types in general over enums. A bit like using interfacs of classes to define data structures in TS. So I am in favor of Marikas change. In most cases you do not need the extra functionality enums provide.

Reply:
Awesome. I am glad to hear that. ;)